examples/azure-service-mesh/security/cert-manager-letsencrypt/readme.md

```diff
@@ -0,0 +1,269 @@
+```console
+NAME                                     READY   STATUS    RESTARTS   AGE
+cert-manager-6fd987499c-xmf44            1/1     Running   0          72s
+cert-manager-cainjector-5b94bd6f-jw644   1/1     Running   0          72s
+cert-manager-webhook-575479ff47-d87pf    1/1     Running   0          72s
+```
+
+### 6. Setup cluster-issuer and Certificate resources
+Set your email address in `cluster-issuer.yaml` that you'd like to register with ACME server.
+```shell
+kubectl apply -f cluster-issuer.yaml
+kubectl apply -f certificate.yaml
+```
+This should create k8s secret `bookinfo-certs` in `aks-istio-ingress` ns as requested by the certificate resource created above.
+
+```shell
+kubectl get secret -n aks-istio-ingress
+```
+
+```console
+NAME                                                                TYPE                 DATA   AGE
+bookinfo-certs                                                      kubernetes.io/tls    2      70s
+sh.helm.release.v1.asm-igx-aks-istio-ingressgateway-external.v112   helm.sh/release.v1   1      3m19s
+sh.helm.release.v1.asm-igx-aks-istio-ingressgateway-external.v113   helm.sh/release.v1   1      80s
+```
+
+### Validate https request to productpage service
+Verify that the productpage can be accessed via the HTTPS endpoint
+
+```shell
+curl -v https://test.dev.azureservicemesh.io/productpage | grep -o "<title>.*</title>"
+```
+this should print `<title>Simple Bookstore App</title>`
+
+## Troubleshooting
+1. Verify that the DNS record has propagated
+```dig +short A test.dev.azureservicemesh.io``` should return the configured ip address ```4.153.8.39``` in the example.
+
+2. check certificate resource readiness
+```shell
+kubectl get certificate -n aks-istio-ingress
+```
+```console
+NAME                         READY   SECRET           AGE
+bookinfo-letsencrypt-certs   False   bookinfo-certs   2m5s
+```
+In case certificate resource is not ready as above, check the resources provisioned by cert-manager in order to solve the ACME challenge. Below is the chain of resources created by cert-manager.
+
+```shell
+kubectl describe certificate bookinfo-letsencrypt-certs -n aks-istio-ingress
+```
+```console
+Name:         bookinfo-letsencrypt-certs
+Namespace:    aks-istio-ingress
+Labels:       <none>
+Annotations:  <none>
+API Version:  cert-manager.io/v1
+Kind:         Certificate
+...
+Events:
+  Type    Reason     Age   From                                       Message
+  ----    ------     ----  ----                                       -------
+  Normal  Issuing    14m   cert-manager-certificates-trigger          Issuing certificate as Secret does not exist
+  Normal  Generated  14m   cert-manager-certificates-key-manager      Stored new private key in temporary Secret resource "bookinfo-letsencrypt-certs-hpjst"
+  Normal  Requested  14m   cert-manager-certificates-request-manager  Created new CertificateRequest resource "bookinfo-letsencrypt-certs-1"
+```
+
+3. check certificaterequest resource.
```

Comment:
what does the user need to check inside this resource?

Reply:
Just wanted to let the users know that these are the chain of resources provisioned by cert-manager and they should troubleshoot these in case something doesn't work for them. Added the header to check for readiness of these resources.